unity/Assets/Scripts/CanOpen_Object.cs

```diff
@@ -97,529 +78,218 @@ void Start ()
-                    "time", animationTime,
-                    "easetype", "linear"));
+                if (openPositions[i].y - closedPositions[i].y != Mathf.Epsilon) {
+                    lerpToPosition.y = ((openPositions[i].y - closedPositions[i].y) * openPercentage) + closedPositions[i].y;
                 }
-
-                //scale with Z axis
-                else if(movementType == MovementType.ScaleZ)
-                {
-                    //we are on the last loop here
-                    if(i == MovingParts.Length -1)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", new Vector3(openPositions[i].x, openPositions[i].y, closedPositions[i].z + (openPositions[i].z - closedPositions[i].z) * openPercentage),
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear", "onComplete", "setisOpen", "onCompleteTarget", gameObject,
-                        "oncompleteparams", openPercentage));
-                    }
-
-                    else
-                    iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                    "scale", new Vector3(openPositions[i].x, openPositions[i].y, closedPositions[i].z + (openPositions[i].z - closedPositions[i].z) * openPercentage),
-                    "islocal", true,
-                    "time", animationTime,
-                    "easetype", "linear"));
+                if (openPositions[i].z - closedPositions[i].z != Mathf.Epsilon) {
+                    lerpToPosition.z = ((openPositions[i].z - closedPositions[i].z) * openPercentage) + closedPositions[i].z;
                 }
-			}
+                args["position"] = lerpToPosition;
+                iTween.MoveTo(MovingParts[i], args);
+            } else if (movementType == MovementType.ScaleY) {
+                args["scale"] = new Vector3(openPositions[i].x, closedPositions[i].y + (openPositions[i].y - closedPositions[i].y) * openPercentage, openPositions[i].z);
+                iTween.ScaleTo(MovingParts[i], args);
+            } else if (movementType == MovementType.ScaleX) {
+                // we are on the last loop here
+                args["scale"] = new Vector3(closedPositions[i].x + (openPositions[i].x - closedPositions[i].x) * openPercentage, openPositions[i].y, openPositions[i].z);
+                iTween.ScaleTo(MovingParts[i], args);
+            } else if (movementType == MovementType.ScaleZ) {
+                args["scale"] = new Vector3(openPositions[i].x, openPositions[i].y, closedPositions[i].z + (openPositions[i].z - closedPositions[i].z) * openPercentage);
+                iTween.ScaleTo(MovingParts[i], args);
+            }
         }
-    }
 
-    private void setisOpen(float openPercentage)
-	{
-        //print("set is open happening");
-		isOpen = !isOpen;
+        setIsOpen(openPercentage: openPercentage);
+    }
 
+    private void setIsOpen(float openPercentage) {
```

Comment:
Here, instead of setting `isOpen` in the metadata to `!isOpen` (i.e., the toggling of open and close), we dynamically set it to `openPercentage != 0`.